docs/ipv6-design-doc.md

```diff
@@ -0,0 +1,164 @@
+    considered; IPv4 addresses are ignored. The patch does not change
+    the behavior of the `use` option, so `use=web` or `use=cmd` can be
+    used for IPv6 if pointed at something that only outputs an IPv6
+    address.
+  * [ddclient-curl](https://github.com/astlinux-project/ddclient-curl)
+    is a fork of ddclient that uses curl as the HTTP client (instead
+    of ddclient's own homemade client) for more robust IPv6 support.
+  * PR #40 is perhaps the most comprehensive attempt at adding full
+    IPv6 support, but it was never merged and has since
+    bit-rotted. There is renewed effort to rebase the changes and get
+    them merged in. PR #40 adds new options and changes some existing
+    options. The approach taken is to completely isolate IPv4 address
+    detection from IPv6 address detection and require the update
+    protocol callbacks to handle each type of address appropriately.
+
+## Requirements
+
+  * The mechanism for determining the current IPv4 address (the `use`
+    option) must be independently configurable from the mechanism used
+    to determine the current IPv6 address.
+  * The user must be able to disable IPv4 address updates without
+    affecting IPv6 updates.
+  * The user must be able to disable IPv6 address updates without
+    affecting IPv4 updates.
+  * If HTTP polling is used for both IPv4 and IPv6 address discovery,
+    the URL used to determine the IPv4 address (the `web` option) must
+    be independently configurable from the URL used to determine the
+    IPv6 address.
+  * The use of IPv4 or IPv6 to update a record must be independent of
+    the type of record being updated (IPv4 or IPv6).
+  * The callback for the update protocol must be given both addresses,
+    even if only one of the two addresses has changed.
+  * The callback for the update protocol must be told which addresses
+    have changed.
+
+### Nice-to-Haves
+
+  * The user should be able to force the update protocol to use IPv4
+    or IPv6.
+  * Unless configured otherwise, ddclient should first attempt to
+    update via IPv6 and fall back to IPv4 if the IPv6 connection
+    fails. This behavior can be added later; for now it is acceptable
+    to keep the current behavior (use IPv6 without IPv4 fallback if
+    there is a AAAA record, use IPv4 if there is no AAAA record).
+  * Full backwards compatibility with existing config files and
+    flags. The trade-off between migration burden and long-term
+    usability should be carefully considered.
+
+## Proposal
+
+### Configuration changes
+
+  * Add new `usev4` and `usev6` settings that are like the current
+    `use` setting except they only apply to IPv4 and IPv6,
+    respectively.
+      * `usev4` can be set to one of the following values: `ipv4`,
+        `webv4`, `fwv4`, `ifv4`, `cmdv4`, `ciscov4`, `cisco-asav4`
+      * `usev6` can be set to one of the following values: `ipv6`,
+        `webv6`, `fwv6`, `ifv6`, `cmdv6`, `ciscov6`, `cisco-asav6`
```

Comment:
I can sign up for ipv6, webv6, ifv6 and cmdv6.  I cannot sign up to implement any of the firewall or gateway methods.